packages/esm-service-queues-app/src/patient-search/visit-form/existing-visit-form.component.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useState } from 'react';
+import { Button, ButtonSet, Form, Row, Stack } from '@carbon/react';
+import { useTranslation } from 'react-i18next';
+import {
+  type ConfigObject,
+  ExtensionSlot,
+  showSnackbar,
+  useConfig,
+  useLayoutType,
+  type Visit,
+} from '@openmrs/esm-framework';
+import { addQueueEntry, useVisitQueueEntries } from '../../active-visits/active-visits-table.resource';
+import { type SearchTypes } from '../../types';
+import styles from './visit-form.scss';
+
+interface ExistingVisitFormProps {
+  toggleSearchType: (searchMode: SearchTypes, patientUuid) => void;
+  visit: Visit;
+  closePanel: () => void;
+}
+
+const ExistingVisitForm: React.FC<ExistingVisitFormProps> = ({ visit, toggleSearchType, closePanel }) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const config = useConfig() as ConfigObject;
+  const visitQueueNumberAttributeUuid = config.visitQueueNumberAttributeUuid;
+  const { mutate } = useVisitQueueEntries('', '');
+
+  if (!visit) {
+    return null;
+  }
+
+  const handleSubmit = useCallback(
+    (event) => {
+      event.preventDefault();
+
+      // retrieve values from queue extension
+      const queueLocation = event?.target['queueLocation']?.value;
+      const serviceUuid = event?.target['service']?.value;
+      const priority = event?.target['priority']?.value;
+      const status = event?.target['status']?.value;
+      const sortWeight = event?.target['sortWeight']?.value;
+
+      setIsSubmitting(true);
+
+      addQueueEntry(
+        visit.uuid,
+        serviceUuid,
+        visit.patient.uuid,
+        priority,
+        status,
+        sortWeight,
+        queueLocation,
+        visitQueueNumberAttributeUuid,
+      ).then(
+        ({ status }) => {
+          if (status === 201) {
+            showSnackbar({
+              kind: 'success',
+              isLowContrast: true,
+              title: t('startAVisit', 'Start a visit'),
+              subtitle: t('queueEntryAddedSuccessfully', 'Queue entry added successfully'),
+            });
+            closePanel();
+            setIsSubmitting(false);
+            mutate();
+          }
+        },
+        (error) => {
+          let subtitle = error?.message;
+          const err = error?.responseBody?.error?.message;
+          if (err && err === '[queue.entry.duplicate.patient]') {
+            subtitle = t('patientAlreadyInQueue', 'Patient is already in the queue');
+          }
```

Comment:
This is my admittedly somewhat hacky solution to the problem presented in https://openmrs.atlassian.net/browse/O3-2624 . In future refactors, we would not get to this point, but would have pre-submission validation in place, but this is an incremental improvement to at least inform the user of the error, which is caught in back end validation with a specific error message returned.